hw/xfree86/common/xf86DGA.c

```diff
@@ -1783,17 +1754,23 @@ ProcXF86DGAGetVideoLL(ClientPtr client)
     /* get the parameters for the mode that best matches */
     DGAGetModeInfo(stuff->screen, &mode, num);
 
+    int dummy_offset;
+    int dummy_flags;
+    CARD32 bank_size = 0;
+    CARD32 offset = 0;
     if (!DGAOpenFramebuffer(stuff->screen, &name,
-                            (unsigned char **) (&rep.offset),
-                            (int *) (&rep.bank_size), &offset, &flags))
+                            (unsigned char **) (&offset),
```

Comment:
This type cast looks fishy. `DGAOpenFramebuffer` will write a pointer into `offset`. And that pointer can be 8 bytes on 64-bit arch? The older code could get away with that since it was all within the structure, but now I'm not sure.

Maybe it would be more save to do something like:
```
unsigned char *offset_ptr;

// call DGAOpenFramebuffer with &offset_ptr

// use ((CARD32)offsetPtr) initialising xXF86DGAGetVideoLLReply
```

Reply:
Btw, this particular method is within `#ifdef DGA_PROTOCOL_OLD_SUPPORT` so maybe we can nuke only that portion for now.

Reply:
that symbol is still defined, so we should do that in a separate step.

Reply:
@algrid I believe that piece of code now fallen off the cliff by DGA-1.0 removal